### PR TITLE
refactor: Introduce ActorSystemError

### PR DESCRIPTION
**Pull request by ark0f**
As preparation for #3025

### CODE REVIEW

File: utils/actor-system-error/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "actor-system-error"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (breathx) [inline utils/actor-system-error/Cargo.toml:6]:
```suggestion
```

**Review by breathx — APPROVED** [review]

File: utils/actor-system-error/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        impl From<$system_err> for $crate::ActorSystemError<$actor_err, $system_err> {
+            fn from(err: $system_err) -> Self {
+                Self::System(err)
+            }
+        }
+    };
+}
+
+#[derive(Debug, Eq, PartialEq, derive_more::Display)]
+pub enum ActorSystemError<A, S> {
```
Comment (clearloop) [inline utils/actor-system-error/src/lib.rs:45]: mb apply `std::error::Error` to this as well?

**Review by clearloop — APPROVED** [review]
docs? 

merge master to trigger the CI ))

File: utils/actor-system-error/src/lib.rs (hunk 2 above)
Comment (ark0f) [reply utils/actor-system-error/src/lib.rs:45]: We mainly work in non-std environment so It's impossible to apply

**Pull request merged**